crypto/core_fetch.c

```diff
@@ -117,7 +117,7 @@ void *ossl_method_construct(OSSL_LIB_CTX *libctx, int operation_id,
         cbdata.force_store = force_store;
         cbdata.mcm = mcm;
         cbdata.mcm_data = mcm_data;
-        ossl_algorithm_do_all(libctx, operation_id, NULL,
+        ossl_algorithm_do_all(libctx, operation_id, prov,
```

Comment:
This seems to do the right thing, but `mcm->get()` below always fetches the "best" algorithm with max score. That will be from default as the query has "?provider=default". Here mcm->get() is actually `get_evp_method_from_store()` which calls `ossl_method_store_fetch()` in property.c. There it does find my signature method with `score = 0` and the one from default with `score=1`. As there is no provider check in there, it picks the one with higher score. 

Picking the higher score one is correct here as we do want to prefer ops from default in general. So, it seems, for this specialized fetch with a provider pointer, one cannot use `mcm->get()` which will call `ossl_method_store_fetch()`.

I'll leave it there and wait for a resolution.

Also to get into those calls I had to bypass some early returns, but not sure whether that's relevant.

Reply:
There's been changes, I assume this bit is resolved.  If not, open another review.

Reply:
> There's been changes, I assume this bit is resolved. If not, open another review.

There is some time-warp going on.. lol. This was an old comment and after that I have tested and commented on commit # 5bee0d4 that it works sometimes but not always.  Now testing # e5e226c